File: problems/58. Length of Last Word/main.go
func lengthOfLastWord(s string) int {
	length := 0
	// Start from the end of the string
	i := len(s) - 1

	// Skip trailing spaces
	for i >= 0 && s[i] == ' ' {
		i--
	}

	// Count characters until we hit a space or the beginning of the string
	for i >= 0 && s[i] != ' ' {
		length++
		i--
	}

	return length
}

File: problems/58. Length of Last Word/main.js
function lengthOfLastWord(s) {
  let length = 0;
  let i = s.length - 1;

  // Skip trailing spaces
  while (i >= 0 && s[i] === " ") {
    i--;
  }

  // Count characters until we hit a space or the beginning of the string
  while (i >= 0 && s[i] !== " ") {
    length++;
    i--;
  }

  return length;
}
